### DISCUSSION TITLE
New boundary creation at intersection of two boundaries

### DISCUSSION THREAD

**Discussion by agniksr** [idaholab/moose#27961] [Q&A Meshing]
So for the given mesh below, i want to create a new boundary called inlet at the intersection of the the top and rmin faces. Similarly, i want to create an outlet boundary at the intersection of the bottom and rmin faces. Could someone please guide me as to how that could be implemented.

```
[Mesh]
   [annular]
     type = AnnularMeshGenerator
     nr = 10
     rmin = 7.5
     rmax = 10.0
     nt = 40
   []
   [make3D]
     type = MeshExtruderGenerator
     extrusion_vector = '0 0 40'
     num_layers = 10
     bottom_sideset = 'bottom'
     top_sideset = 'top'
     input = annular
   []
[]
```

**Comment by GiudGiud** [comment]
Hello

You'll want to use the AdvancedExtruderGenerator:
https://mooseframework.inl.gov/source/meshgenerators/AdvancedExtruderGenerator.html
You can use the downward_boundary_ids and upward_boundary_ids to create these sidesets inside the extrusion

Guillaume

**Reply by agniksr** [reply]
Thanks for the suggestion, however, when i try to use that as below
```
[Mesh]
   [annular]
     type = AnnularMeshGenerator
     nr = 10
     rmin = 7.5
     rmax = 10.0
     nt = 40
   []
   [extrude]
    type = AdvancedExtruderGenerator
    input = annular
    heights = '1'
    num_layers = 10
    direction = '0 0 40'
    bottom_boundary = '100'
    top_boundary = '200'
    upward_boundary_source_blocks = '0'
    upward_boundary_ids = '1001'
    downward_boundary_source_blocks = '0'
    downward_boundary_ids = '1501'
  []
[]
```
it does not give me  the inner boundary, instead it just gives me the bottom of the lower boundary and the top of the lower bounadry. How exactly can i change it to give only the inner radial elements at the top and bottom boundaries.

**Reply by GiudGiud** [reply]
ok I misunderstood what you wanted.
Let's use a ParsedGenerateSideset
https://mooseframework.inl.gov/source/meshgenerators/ParsedGenerateSideset.html
inner radial elements should obey an analytical parsed expression you can specify in the `combinatorial_geometry` parameter
